### ISSUE TITLE
Question: expected behavior for session closed by router

### ISSUE THREAD

**Issue by mjentsch** [gammazero/nexus#194]
Hi, I'm new to WAMP and made some first steps using the provided examples/simple code... if I start the server, then the subscriber and then stop the server, the subscriber exists with the message "Router gone, exiting" - that's what I would expect. Yet if I do the test with the publisher (I put publisher.Publish() in a loop) it does not exit but seems to continue to publish... weird but something I should expect?

Thanks

**Comment by gammazero** [comment]
The publish behavior is a result of the client putting the message on a channel for the transport to handle delivery of, without checking to see if the transport is still handling messages.  This code assumed that if you cared about determining if the event was publish, then you would use the publish-acknowledge feature:

If you requested publish acknowledgement,
```
    opts := wamp.Dict{wamp.OptAcknowledge: true}
    err = publisher.Publish(exampleTopic, opts, args, nil)
```
you would have seen an error: `timeout waiting for reply`

That being said... it is weird; not what someone would probably expect.  That means that people may likely have an event publisher that publishes in a loop without detecting the loss of the router.

I have created PR #195 to fix this.  The client will check the connected status before sending a message.  It is still possible that a message is flight never arrives at a disconnected router (for critical messages use publish-acknowledge feature), but repeated publishing with a lost router connection will not happen.

**Comment by mjentsch** [comment]
Thanks for that, I get a "client not connected" with this update.
I see that I can check for the connection with the channel returned by (*Client).Done() which helps with long connected publishers, checking that before the Publish call results in similar behavior as with the subscriber. So I wouldn't call it an issue if you don't pull in your update :)

**Comment by gammazero** [comment]
Fixed by #195
